src/dataset/impl_dataset.rs

```diff
@@ -153,6 +166,20 @@ impl<L, R: Records, T: AsTargets<Elem = L>> DatasetBase<R, T> {
             targets: targets.map(fnc),
             weights,
             feature_names,
+            target_names,
+        }
+    }
+
+    /// Returns target names
+    ///
+    /// A target name gives a human-readable string describing the purpose of a single target.
+    pub fn target_names(&self) -> Vec<String> {
+        if !self.target_names.is_empty() {
+            self.target_names.clone()
+        } else {
+            (0..self.ntargets())
+                .map(|idx| format!("class-{}", idx))
+                .collect()
```

Comment:
If there are no names, this method should instead just return empty list. The signature should be `fn target_names(&self) -> &[String]` to prevent cloning. Apply this change to `feature_names` as well for consistency.